cluster/images/etcd/migrate/rollback_v2.go

```diff
@@ -97,43 +88,43 @@ func main() {
 		return nil
 	})
 	if err != nil {
-		glog.Fatal(err)
+		return err
 	}
 	tx.Unlock()
 
 	if err := traverseAndDeleteEmptyDir(st, "/"); err != nil {
-		glog.Fatal(err)
+		return err
 	}
 
 	// rebuild cluster state.
-	metadata, hardstate, oldSt, err := rebuild(*migrateDatadir)
+	metadata, hardstate, oldSt, err := rebuild(migrateDatadir)
 	if err != nil {
-		glog.Fatal(err)
+		return err
 	}
 
 	// In the following, it's low level logic that saves metadata and data into v2 snapshot.
-	backupPath := *migrateDatadir + ".rollback.backup"
-	if err := os.Rename(*migrateDatadir, backupPath); err != nil {
-		glog.Fatal(err)
+	backupPath := migrateDatadir + ".rollback.backup"
+	if err := os.Rename(migrateDatadir, backupPath); err != nil {
+		return err
 	}
-	if err := os.MkdirAll(path.Join(*migrateDatadir, "member", "snap"), 0700); err != nil {
-		glog.Fatal(err)
+	if err := os.MkdirAll(path.Join(migrateDatadir, "member", "snap"), 0700); err != nil {
+		return err
 	}
-	walDir := path.Join(*migrateDatadir, "member", "wal")
+	walDir := path.Join(migrateDatadir, "member", "wal")
 
 	w, err := oldwal.Create(walDir, metadata)
 	if err != nil {
-		glog.Fatal(err)
+		return err
 	}
 	err = w.SaveSnapshot(walpb.Snapshot{Index: hardstate.Commit, Term: hardstate.Term})
 	if err != nil {
-		glog.Fatal(err)
+		return err
 	}
 	w.Close()
```

Comment:
Are we deliberately trying to skip close() in the event of an error saving a snapshot? That seems wrong...

Reply:
Good catch, this is what we run in production today :P  Fixing.